File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> res(nums.size(),1);
        int p=1;
        for(int i=0;i<nums.size();i++){
            res[i]=p;
            p=p*nums[i];
        }

       int q=1;
        for(int j=nums.size()-1;j>=0;j--){
            res[j]=res[j]*q;
            q=q*nums[j];
        }
        return res;
    }
};

File: Data Structures & Algorithms/eating-bananas/submission-0.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l,r = 1 , max(piles)
        result = r
        while l <= r:
            k = (l + r) // 2 # // gives floor as ans like 2.63 if res of divison it will ans 2
            totalTime = 0
            for i in piles:
                totalTime += math.ceil(i / k) # / gives floated ans fill 2.63
            if totalTime <= h:
                res = k
                r = k - 1
            else:
                l = k + 1
        return res
